plugins/plugin_utils/lookup.py

```diff
@@ -124,9 +124,9 @@ def get_credentials(**options):
     credentials["vcenter_password"] = options.get("vcenter_password") or os.getenv(
         "VMWARE_PASSWORD"
     )
-    credentials["vcenter_validate_certs"] = options.get(
-        "vcenter_validate_certs"
-    ) or os.getenv("VMWARE_VALIDATE_CERTS")
+    credentials["vcenter_validate_certs"] = options.get("vcenter_validate_certs")
```

Comment:
According to this doc_fragment https://github.com/ansible-collections/vmware.vmware_rest/blob/main/plugins/doc_fragments/moid.py#L46 vcenter_validate_certs defaults true if not set. As of now, is this thing verified?

Reply:
Yes, `vcenter_validate_certs` defaults to `true` if not set either in a playbook var or in env. That happens in the base `AnsiblePlugin.set_options()` function  _because_ the default is `true` in the doc_fragment. This fix doesn't change any of that, it just ensures that an intentional `false` option passed to the module by the user is not overwritten by the env variable.

Reply:
This is weird... if `vcenter_validate_certs` has a default, it should _never_ fall back to an environment variable. But the documentation says it does.

What I mean is that the fallback should take place when `vcenter_validate_certs` isn't set, but it's _always_ set because of the default value. Because of the default, it's _never_ unset.

There might be a very basic problem here... or am I wrong?

Reply:
I think perhaps the documentation here is just worded in a confusing way. What happens (and what I think the documentation is trying to convey) is that if a user passes the `vcenter_validate_certs` option, that takes priority even if the env variable is also set. If the option is not passed, it checks for the env variable, and then if there is no option or env variable it defaults to `true`.

Reply:
if you use the native config system, this is not only automatic, but well documented and would also show the lookup documentation natively https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#plugin-configuration-documentation-standards

Reply:
@bcoca ah so we don't even need this `get_credentials()` function? Even better :)

Reply:
> Yes, `vcenter_validate_certs` defaults to `true` if not set either in a playbook var or in env. That happens in the base `AnsiblePlugin.set_options()` function _because_ the default is `true` in the doc_fragment. This fix doesn't change any of that, it just ensures that an intentional `false` option passed to the module by the user is not overwritten by the env variable.

I see that the change is not affecting the original behavior, but since your patch is supposed to fix the `vcenter_validate_certs` issue, I thought you should see if the documented behavior is consistent.

I guess you should document the env var in the doc_fragments if using the approach @bcoca suggested.